aten/src/ATen/native/NamedTensor.cpp

```diff
@@ -16,5 +16,141 @@ Tensor set_names(const Tensor& self, optional<DimnameList> names) {
   return result;
 }
 
+static void report_moving_unnamed_dim_error(
+    DimnameList names, DimnameList other, bool is_aligning_two_tensors) {
+  if (is_aligning_two_tensors) {
+    TORCH_CHECK(false,
+        "Aligning Tensor", names, " and Tensor", other,
+        " would change the absolute position from the right of an unnamed dimension. ",
+        "Please name unnamed dimensions to avoid ambiguity.");
+  } else {
+    TORCH_CHECK(false,
+        "Aligning Tensor", names, " to `names` ", other,
```

Comment:
I think you're missing a space car after 'Tensor'

Reply:
This was intentional. The error message will look like
```
Aligning Tensor['N', 'C', 'H', 'W'] to Tensor['W', 'H'] ...
```

Reply:
ah OK